Dockerfile.test

```diff
@@ -0,0 +1,57 @@
+# This Docker image contains a minimal build environment for TiKV
```

Comment:
What's the difference between this and Dockerfile? `nextest` is not needed for `make test`.

Reply:
> What's the difference between this and Dockerfile? `nextest` is not needed for `make test`.

No major difference. But it's a pure dev environment and the Dockerfile includes the run-time image and it also does not include the tests folder. 
This docker file aims to provide a quick way for development and test in docker environment. And I know nextest is not needed for make test. But people can choose to use make test_with_nextest if they want manually launch the tikv_dev container. So it's better to make it ready for use.